app/controllers/api/admin/articles_controller.rb

```diff
@@ -9,6 +9,21 @@ def index
       render json: article, each_serializer: Admin::Article::IndexSerializer
   end
 
+  def update
+    if params[:activity] == "PUBLISH"
+      begin
+        article = Article.find(params[:id])
+        article.premium = params[:premium] || article.premium
+        article.category = params[:category] || article.category
+        article.published = true
+        article.save
+        render json: { message: 'Article successfully published!'}
```

Comment:
We should probably have a `if article.persisted?` here?

Reply:
We considered doing a if article.save (.persisted? is for creation I think, and will probably return true regardless of if .save worked or not), but we are inside a begin rescue, so any failure to save will be thrown as error, we are testing a lot for this. It is not needed imo to have a save check.

Reply:
Ah, I see. That makes sense.